org.freedesktop.Sdk.Extension.gnat13.yml

```diff
@@ -0,0 +1,71 @@
+# SPDX-FileCopyrightText: 2024 Junde Yhi <junde@yhi.moe>
+# SPDX-License-Identifier: CC0-1.0
+
+id: org.freedesktop.Sdk.Extension.gnat13
+branch: '23.08'
+runtime: org.freedesktop.Sdk//23.08
+sdk: org.freedesktop.Sdk//23.08
+build-extension: true
+separate-locales: false
+appstream-compose: false
+modules:
+  - name: extract-gnat
+    buildsystem: simple
+    build-commands:
+      - cp -R * /usr/lib/sdk/gnat13
+    sources:
+      - type: archive
+        url: https://github.com/alire-project/GNAT-FSF-builds/releases/download/gnat-13.2.0-1/gnat-x86_64-linux-13.2.0-1.tar.gz
```

Comment:
It won't work, I should add `only_arches` to it. I realised that there are more relevant documentation that I didn't notice. I just reviewed [For app authors \> Runtime](https://docs.flathub.org/docs/for-app-authors/runtimes); because not all builds on [alire-project/GNAT-FSF-builds](https://github.com/alire-project/GNAT-FSF-builds) are on aarch64 and I haven't tried building from source yet, I can either mark this package x86_64 only or continue rewriting the manifest so that it builds from source. Do you think temporarily marking this x86_64 only is a choice?

Reply:
how about building from source?

Reply:
It's my first time on the both (Flatpak \& GNAT), I only tried to make it work on my machine, so I'm not quite sure at this moment, but I'll try.

Reply:
I am just chiming in.

GNAT, sadly or luckily, is written in Ada, which means that a previously installed GNAT is needed in order to compile it. I do not know if Flatpaks are run in a system where GNAT (gcc-ada) is installed and can be built, but afaik, there is no other Flatpak that has an Ada compiler available. The compilation toolchain used here is built from source using an automated script that uses Github runners to install the distribution's Ada toolchain as a base, and then builds the final GNAT binaries, so it should be transparent. These binaries are the ones used by Alire, which is just Cargo or Pip but for Ada, so they are fairly widely used.

Bootstrapping Ada is a pain, I am not going to lie here about that :/

Reply:
Regarding aarch64 compatibility, there are GNAT binaries available for aarch64 from another community member using the same build method as the one above. They can be found here https://github.com/reznikmm/GNAT-FSF-builds/releases